net/keepalived/files/keepalived.init

```diff
@@ -2,13 +2,355 @@
+	for opt in virtual_ipaddress virtual_ipaddress_excluded; do
+		config_get $opt $1 $opt
+		eval optval=\$$opt
+		[ -z "$optval" ] && continue
+		printf "$INDENT_1$opt {\n" >> $KEEPALIVED_CONF
+		for a in $optval; do
+			config_foreach print_ipaddress_indent ipaddress $a $INDENT_2
+		done
+		printf "$INDENT_1}\n" >> $KEEPALIVED_CONF
+	done
+
+	# Handle virtual_routes
+	for opt in virtual_routes; do
+		config_get $opt $1 $opt
+		eval optval=\$$opt
+		[ -z "$optval" ] && continue
+		printf "$INDENT_1$opt {\n" >> $KEEPALIVED_CONF
+		for r in $optval; do
+			config_foreach print_route_indent route $r $INDENT_2
+		done
+		printf "$INDENT_1}\n" >> $KEEPALIVED_CONF
+	done
+
+	# Handle track_interface & track_script lists
+	for opt in track_interface track_script; do
+		config_get $opt $1 $opt
+		eval optval=\$$opt
+		[ -z "$optval" ] && continue
+		printf "$INDENT_1$opt {\n" >> $KEEPALIVED_CONF
+		for t in $optval; do
+			config_foreach print_track_elem_indent $opt $t $INDENT_2
+		done
+		printf "$INDENT_1}\n" >> $KEEPALIVED_CONF
+	done
+
+	# Handle simple lists of strings (with no spaces in between)
+	for opt in unicast_peer; do
+		config_get $opt $1 $opt
+		print_list_indent $opt
+	done
+	unset optval
+
+	config_get auth_type $1 auth_type
+	config_get auth_pass $1 auth_pass
+	[ -n "$auth_type" -a -n "$auth_pass" ] && {
+		printf "${INDENT_1}authentication {\n" >> $KEEPALIVED_CONF
+		printf "${INDENT_2}auth_type $auth_type\n" >> $KEEPALIVED_CONF
+		printf "${INDENT_2}auth_pass $auth_pass\n" >> $KEEPALIVED_CONF
+		printf "$INDENT_1}\n" >> $KEEPALIVED_CONF
+	}
+
+	print_elems_indent $1 $INDENT_1 use_vmac state interface \
+		mcast_src_ip unicast_src_ip virtual_router_id version priority \
+		advert_int preempt_delay debug notify_backup \
+		notify_master notify_fault notify_stop notify \
+		lvs_sync_daemon_interface garp_master_delay garp_master_refresh \
+		garp_master_repeat garp_master_refresh_repeat \
+		no_val_vmac_xmit_base no_val_native_ipv6 no_val_accept \
+		no_val_dont_track_primary no_val_smtp_alert no_val_nopreempt
+}
+
+vrrp_script() {
+	local name
+
+	config_get name $1 name
+	[ -z "$name" ] && return 0
+
+	config_section_open "vrrp_script" "$name"
+
+	print_elems_indent $1 $INDENT_1 script interval weight fall rise
+}
+
+process_config() {
+	# First line
+	printf "! Configuration File for keepalived (autogenerated via init script)\n\n" > $KEEPALIVED_CONF
+
+	config_load 'keepalived'
+	config_foreach_wrapper global_defs
+	config_foreach_wrapper static_ipaddress
+	config_foreach_wrapper static_routes
+	config_foreach_wrapper vrrp_sync_group
+	config_foreach_wrapper vrrp_sync_instance
+	config_foreach_wrapper vrrp_script
+}
+
+start_service() {
+	procd_open_instance
+	procd_set_param command /usr/sbin/keepalived
+	procd_append_param command -n # don't daemonize, procd will handle that for us
+	procd_append_param command -f "$KEEPALIVED_CONF"
+
+	if [ -f /etc/keepalived.conf ]; then
+		cp -f /etc/keepalived.conf $KEEPALIVED_CONF
```

Comment:
i'm ok to move it but:
it must be transparent to the user (post install hook or ...)
and
you must keep it on upgrades (conffiles sections)